sphinx/domains/std.py

```diff
@@ -780,7 +780,9 @@ def process_doc(self, env: "BuildEnvironment", docname: str, document: nodes.doc
             self.labels[name] = docname, labelid, sectname
 
     def add_program_option(self, program: str, name: str, docname: str, labelid: str) -> None:
-        self.progoptions[program, name] = (docname, labelid)
+        # prefer first command option entry
+        if (program, name) not in self.progoptions:
+            self.progoptions[program, name] = (docname, labelid)
```

Comment:
Why is this bit needed?

Reply:
If you a duplicate option directive like:

```rst
.. option:: -Wimplicit-fallthrough

   Warn when a switch case falls through. Legal values are:

.. option:: -Wimplicit-fallthrough=3

   Test
```

Then all cross-refs (e.g. `:option:`-Wimplicit-fallthrough=n` would point to the first directive and not the second one. Hope it's desired behavior.

Reply:
Thanks for clarifying